lib/active_merchant/billing/gateways/shift4_v2.rb

```diff
@@ -57,6 +57,34 @@ def add_amount(post, money, options, include_currency = false)
         super
         post[:currency]&.upcase!
       end
+
+      def add_creditcard(post, payment_method, options)
+        return super unless payment_method.is_a?(Check)
+
+        post.merge!({
+          paymentMethod: {
+            type: :ach,
+            fraudCheckData: { ipAddress: options[:ip] },
```

Comment:
based upon on the [documentation](https://dev.shift4.com/docs/api#payment-method-create_fraudCheckData) since we are sending `:external` in the `verificationProvider`, the `ipAddress` is not required, we can avoid to send it or add validation to ensure it is sent only if verificationProvider == 'none'.

Reply:
Yes, I tried, maybe is because the special change they made for us, if remove that param then I get this error:

![Screenshot 2024-04-09 at 3 30 40 PM](https://github.com/activemerchant/active_merchant/assets/133580/4043b6c6-04f5-4460-b75b-d042f218edba)

Reply:
💭 Opinion:
This doesn't mean it can accept another value, not necessarily the ipAddress, which could be the email.